clients/java/src/main/java/io/unitycatalog/client/ApiClientBuilder.java

```diff
@@ -84,22 +96,24 @@ public ApiClientBuilder tokenProvider(TokenProvider tokenProvider) {
    * Sets application version metadata as name-version pairs.
    *
    * <p>Arguments must be provided in alternating name-version order with an even count, e.g.,
-   * {@code "MyApp1", "0.1.0", "MyApp2", "1.0.2"}.
+   * {@code "MyApp1", "0.1.0"}.
    *
-   * @param nameVersionPairs the metadata as alternating name-version pairs
+   * @param name the application name.
+   * @param version the application version.
    * @return this builder instance for method chaining
-   * @throws IllegalArgumentException if an odd number of arguments is provided
+   * @throws IllegalArgumentException if any argument is null or empty.
    */
-  public ApiClientBuilder appVersion(String... nameVersionPairs) {
+  public ApiClientBuilder addAppVersion(String name, String version) {
```

Comment:
The `addAppVersion(name, version)` is more simple than the `appVersion(String...nameVersionPairs)`, since we can easily to check every `name` and `version` when call addAppVersion method, report the exact error message for every call. 

While using the `String[]`, is actually hard to report the clear error message if there is any argument checking invalidation.

Reply:
the downside of this style of API is that we can never remove anything once added. 
which is okay in this case.. but just so you know.

Reply:
Agreed, for builder we don't need to remove anything.  But for other cases,  it may not the best choice.